tests/metrics_test.go

```diff
@@ -153,14 +158,104 @@ var _ = Describe("Metrics", func() {
 		)
 	})
 
+	Context("SSP metrics", func() {
+		var originalVersion string
+
+		getDeployment := func(name string, namespace string) *apps.Deployment {
+			deployment := &apps.Deployment{}
+			err := apiClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, deployment)
+			Expect(err).ToNot(HaveOccurred())
+			return deployment
+		}
+
+		updateVersion := func(version string) {
+			Eventually(func() error {
+				deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
+
+				envs := deployment.Spec.Template.Spec.Containers[0].Env
+				var newEnvs []k8sv1.EnvVar
+
+				for _, env := range envs {
+					if env.Name != common.OperatorVersionKey {
```

Comment:
Is there a reason why the existing `EnvVar` cannot be changed in-place?

Reply:
how so?

Reply:
I meant like this:

```
for i, env := range envs {
    if env.Name == common.OperatorVersionKey {
        envs[i].Version = version
        break
    }
}
```

Also, does the update need the surrounding `Eventually`?

Reply:
It's a good practice to have `Eventually` around update, because the resource in the cluster may be concurrently updated by another component. Then update would fail.